### DISCUSSION TITLE
EVENTS IN OBJECT

### DISCUSSION THREAD

**Discussion by cho66x** [naverz/zepeto-studio-global#34] [General]
Hello Zepeto Tech Evangelist, I want to ask, how to create events in an object. for example: "avatar wants to get on a boat" when he clicks the ui button, he automatically gets on the boat

**Comment by jhslee** [comment]
Hello @anggacho 👋 

Here's a quick plugin I put together with the interaction/event system. This should cover that UI button / interaction system you wanted to create. Please feel free to explore the scripts and modify any files as you desire. The assets in there, including the sitting animation are free for you to use. 

[Interaction Plugin](https://github.com/naverz/zepeto-studio-global/files/9293997/InteractionPlugin_V1.unitypackage.zip)
[Plugin Usage Walkthrough](https://youtu.be/JVFzUX_VJoM)

Please let me know if you have any questions!! 
John

**Reply by ckdlskgkdrhd** [reply]
thanks for the video, cheers!

**Reply by ckdlskgkdrhd** [reply]
@jhslee I've went and recreate what you've shown on the video, but I'm facing a certain problem at the end.
despite having to rewatch the video to spot my mistake, I've failed to notice any difference. 
![22](https://user-images.githubusercontent.com/110361648/183793176-55e57ff2-28ff-4750-a72d-d9ebc8ff9fb9.PNG)

**Reply by ckdlskgkdrhd** [reply]
nvm, I forgot to add in interaction manager. lol

**Reply by ckdlskgkdrhd** [reply]
so.. everything works until I try to move my character off the cube now. 
![aa](https://user-images.githubusercontent.com/110361648/183796323-3558496c-0d2a-4c73-b746-fa06a712edba.PNG)

**Reply by ckdlskgkdrhd** [reply]
alright, I got it all working perfectly.
I had to change the no.34th line on InteractionTrigger.ts
from: const exit = enter.transform.Find("ExitPoint");
to : const exit = enter.transform.Find("Exit");
idk how you got it working on your video tho.

**Reply by jhslee** [reply]
Thats awesome! Yes, sorry about that, please ensure to name the exit object "ExitPoint" and not exit, that was my mistake in the video. @ckdlskgkdrhd 's suggestion will work just the same way too 🙂.

**Reply by padorami** [reply]
I want to make 100 cubes to sit on. But if I use that script to create 100 cubes with each collider, it stays at the DOCK POINT of the first cube. How should I modify the script?

**Comment by cho66x** [comment]
Thanks u very much @jhslee
